app/routes/user.js

```diff
@@ -0,0 +1,9 @@
+import express from 'express';
+
+import UserController from '../controllers/userController';
+
+const Router = express.Router();
+
+Router.post('/signup', UserController.signup) 
```

Comment:
Instead of doing validation in the controllers, lets have a validation middleware that calls next if validation passes

Reply:
Makes sense. But what advantage does it give over having it as a helper function? 

I do understand that "ideally"  it's not the work of the controller to handle the business logic, like the `validation` in this case... But our current structure doesn't have `services` which should handle the business logic.

Reply:
> I do understand that "ideally" it's not the work of the controller

This is why, every dev looking at it would expect validations to be part of a middleware... and not having the controller doing so much work leads to the code being more cleaner and readable.

> But our current structure doesn't have services which should handle the business logic

Structure should not affect how validation should be done